src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -316,8 +414,6 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 	{
```

Comment:
I accept that the name is a bit misleading. I renamed it to ConvertToDistributedTable(). Also, I agree that the refactoring you propose makes sense. On the other hand, this code part was already inside a function, and changes in this PR already have become too big. If you don't have strong objections, I prefer to leave refactoring of this existing code part outside of this PR.